Improve TemporaryJob opaque err msg and don't let user set name

Petter tried to start an elastic search tempjob

```java
    log.info("starting elastic search");
    elasticSearch = temporaryJobs.job()
        .image("elasticsearch:latest")
        .name("elasticsearch")
        .port("tcp", 9300)
//        .registration("elasticsearch", "tcp", "tcp")
        .deploy();
```
It fails with

```
java.lang.AssertionError: Failed to create job elasticsearch:null - CreateJobResponse{status=INVALID_JOB_DEFINITION, errors=[Job hash was not specified in job id [elasticsearch:null:fd23f2e0da5c96f1bef0fa6a32f005c4ed90edcd].], id=elasticsearch:null:9620f7ed8ca9ef4785474b1ddf46057652a39db6}
    at org.junit.Assert.fail(Assert.java:88)
    at com.spotify.helios.testing.TemporaryJob.deploy(TemporaryJob.java:174)
    at com.spotify.helios.testing.DefaultDeployer.deploy(DefaultDeployer.java:135)
    at com.spotify.helios.testing.DefaultDeployer.deploy(DefaultDeployer.java:95)
    at com.spotify.helios.testing.TemporaryJobBuilder.deploy(TemporaryJobBuilder.java:287)
    at com.spotify.helios.testing.TemporaryJobBuilder.deploy(TemporaryJobBuilder.java:248)
    at com.spotify.sys.acceptance.World.startElasticSearch(World.java:123)
    at com.spotify.sys.acceptance.AcceptanceIT.startUp(AcceptanceIT.java:13)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
    at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
    at cucumber.api.junit.Cucumber.run(Cucumber.java:87)
    at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
    at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:78)
    at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:212)
    at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:68)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
docker run -d elasticsearch works without problems. I get the same error when running mvn verify from the command line as I do when running the tests from inside IDEA.
```
> Any ideas what the problem is? Any ideas about how to help improve the error message? It’s not super-user-friendly to tell me that I didn’t specify a job hash, when the API I’m using doesn’t seem to allow me to specify a job hash - and I’m a helios n00b, so I don’t really know what the job hash is.

@rohansingh said

> Could you try getting rid of this line:

```java
        .name("elasticsearch")
```
> I think that should fix it. Guessing the problem is that TemporaryJobs calculates its own job names & hash, and by setting your own job name, the hash is no longer valid.

> Definitely an unclear error message, and also a bug that you can do this.

--- a/helios-testing/src/main/java/com/spotify/helios/testing/TemporaryJobBuilder.java
+++ b/helios-testing/src/main/java/com/spotify/helios/testing/TemporaryJobBuilder.java
@@ -89,11 +89,6 @@ public class TemporaryJobBuilder {
     this.reportWriter = reportWriter;
   }
 
-  public TemporaryJobBuilder name(final String jobName) {
-    this.builder.setName(jobName);
-    return this;
-  }
-
   public TemporaryJobBuilder version(final String jobVersion) {
     this.builder.setVersion(jobVersion);
     return this;
@@ -282,7 +277,6 @@ public class TemporaryJobBuilder {
         waitPorts.clear();
       }
 
-      boolean success = false;
       if (this.hosts.isEmpty()) {
         if (isNullOrEmpty(hostFilter)) {
           hostFilter = env.get("HELIOS_HOST_FILTER");

--- a/helios-testing/src/main/java/com/spotify/helios/testing/TemporaryJobs.java
+++ b/helios-testing/src/main/java/com/spotify/helios/testing/TemporaryJobs.java
@@ -217,9 +217,6 @@ public class TemporaryJobs implements TestRule {
       }
     }
 
-    if (config.hasPath("name")) {
-      builder.name(config.getString("name"));
-    }
     if (config.hasPath("version")) {
       builder.version(config.getString("version"));
     }

